src/AnalyzeView/GeoTagController.cc

```diff
@@ -8,120 +8,172 @@
  ****************************************************************************/
 
 #include "GeoTagController.h"
+#include "GeoTagWorker.h"
 #include "QGCLoggingCategory.h"
 
 #include <QtCore/QDir>
+#include <QtCore/QFileInfo>
+#include <QtCore/QThread>
 #include <QtCore/QUrl>
 
 QGC_LOGGING_CATEGORY(GeoTagControllerLog, "qgc.analyzeview.geotagcontroller")
 
-GeoTagController::GeoTagController()
-    : _progress(0)
-    , _inProgress(false)
+GeoTagController::GeoTagController(QObject *parent)
+    : QObject(parent)
+    , _worker(new GeoTagWorker())
+    , _workerThread(new QThread(this))
 {
-    connect(&_worker, &GeoTagWorker::progressChanged,   this, &GeoTagController::_workerProgressChanged);
-    connect(&_worker, &GeoTagWorker::error,             this, &GeoTagController::_workerError);
-    connect(&_worker, &GeoTagWorker::started,           this, &GeoTagController::inProgressChanged);
-    connect(&_worker, &GeoTagWorker::finished,          this, &GeoTagController::inProgressChanged);
+    // qCDebug(GeoTagControllerLog) << Q_FUNC_INFO << this;
+
+    _worker->moveToThread(_workerThread);
+
+    (void) connect(_worker, &GeoTagWorker::progressChanged, this, &GeoTagController::_workerProgressChanged);
```

Comment:
Is there something new going on with connect which required the (void) now?

Reply:
It's just a pattern carried over from my embedded work. I like to denote that I am intentionally ignoring return values rather than being unaware if it has one or not, helps with collaborating so others know that too